src/vm/eventpipeconfiguration.cpp

```diff
@@ -47,8 +49,11 @@ EventPipeConfiguration::~EventPipeConfiguration()
 
     if(m_pProviderList != NULL)
     {
-        // Take the lock before manipulating the provider list.
-        CrstHolder _crst(EventPipe::GetLock());
+        // Here we manipulate the list without taking a lock
```

Comment:
I'm don't think it's safe to manipulate this list without taking the lock.  The concern is that if there is some code that happens to be adding or deleting a provider, then it will be in a race with this code.  I know that this happens on shutdown right now, and so the chances of that are low, but it is still possible.

Rather than not taking the lock, I recommend putting all of this in a try/catch block and if the lock gets broken by the host then just let it throw and continue on shutdown without finishing the clean-up.  This won't affect any tracing sessions and will just leak the memory.  On shutdown, this isn't an issue, and if we choose to change the lifetime of the configuration, say for supporting multiple sessions, then we'll absolutely need the lock (though we might also eventually decide to have a per-session lock, but we'll decide that when we get there...)